### ISSUE TITLE
Consider extending the VS Code API with a method for capturing window screenshots

### ISSUE THREAD

**Issue by TomasHubelbauer** [Microsoft/vscode#49959]
_I have searched the issues for *screenshot*, but unfortunately GitHub search gave me about 120 pages of unrelated results due to people attaching screenshots to their issues… Google search didn't help either, so I assume this has not been asked before, if yes, please excuse this duplicate._

**I'd like to request for the VS Code API to be extended with a `window.captureScreenshot` method.**

## The Reasoning

The reason I request this is that I'd like to automate generating screenshots of my VS Code extension in action as it runs in CI.

Electron already has the `window.webContents.capturePage` API an the VS Code API offers enough methods to manipulate views, editors, selections etc. to stage the scene for the screenshot.

## The Scope

Since extensions run in separate processes and do not have access to the broader Electron API, it is not possible for them to invoke `window.webContents.capturePage` directly, that's why I think it is necessary to expose it through the VS Code API.

## The Alternative

Initially I considered automating some existing cross-platform screen capturing software and automation software on a build server which would have to have a graphical interface and a window manager, but then figured it makes sense to use VS Code directly as it can expose the Electron API in a safe way and already contains enough API methods for "automation".

**Comment by jrieken** [comment]
> does the test runner have access to the Electron renderer process? 

@bpasero would know

**Comment by TomasHubelbauer** [comment]
So @bpasero can you think of a good way to automate the screenshot capturing while somehow utilizing VS extensibility so it's not needed to try and stitch this together using desktop environments in the CI system? Maybe even a hacky way to obtain the Electron rendered/main access from knowing the window handle or something like that? Remotely attach debugger to VS Code while the extension host is running? Something like that?

**Comment by TomasHubelbauer** [comment]
Is there a way to instruct the extension test runner to start up VS Code extension host window with `--inspect` so that the V8 inspector protocol could be used (I'm sure there are Node packages for this)?

**Comment by TomasHubelbauer** [comment]
I found out that just adding arguments to `yarn test` won't them to arguments passed to the VS Code extension host. I do not want to edit `node_modules/vscode/.bin` so I decided to try and attach a debugger to a running VS Code extension host window instance from the test code using `process._debugProcess` from Node (equivalent of the Unix syscall `SIGUSR1` which attaches debugger to running Node process), but it doesn't seem to work for Electron for Windows as tracked in electron/electron#13580

**Comment by TomasHubelbauer** [comment]
I just noticed that the Debug Console pane shows the inspector port and target in the output when initiating the Extension Tests launch configuration. I try to use CDP to connect to it given the values from that output line, but it failed saying it was not an inspectable target. Sure enough, `chrome://inspect` didn't show it either. I also figured I could just pass `--inspect` using the debug configuration `args` so I did that, but I wasn't successful in using CDP to connect to that port either.
